src/WebJobs.Extensions.DurableTask/Bindings/ActivityTriggerAttributeBindingProvider.cs

```diff
@@ -86,11 +81,47 @@ private class ActivityTriggerBinding : ITriggerBinding
                 this.parameterInfo = parameterInfo;
                 this.attribute = attribute;
                 this.activityName = activity;
+                this.contract = GetBindingDataContract(parameterInfo);
             }
 
             public Type TriggerValueType => typeof(DurableActivityContext);
 
-            public IReadOnlyDictionary<string, Type> BindingDataContract => StaticBindingContract;
+            public IReadOnlyDictionary<string, Type> BindingDataContract => this.contract;
+
+            private static IReadOnlyDictionary<string, Type> GetBindingDataContract(ParameterInfo parameterInfo)
+            {
+                var contract = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase)
+                {
+                    // This binding supports return values of any type
+                    { "$return", typeof(object).MakeByRefType() },
+                    { nameof(DurableActivityContext.InstanceId), typeof(string) },
+                };
+
+                if (IsSimple(parameterInfo.ParameterType))
```

Comment:
What is this section doing? The reflection code in IsSimple is a bit scary

Reply:
The intent is to check and see if the input parameter is a "simple type" that can be used for binding - e.g. a primitive, string, decimal or a nullable of one of these types. The main thing I want to avoid is allowing users to bind to complex types which can't easily be converted to a meaningful string value.  Do you think this check is overkill? I wasn't sure what the underlying binding infrastructure does if a user tries to use a random complex type in a binding expression.

Reply:
I'd prefer we skip the check here (it's very policy specific); and can push it down to the SDK level (so that it's consistent and all the extensions are doing the same thing).